action/remove.go

```diff
@@ -2,24 +2,70 @@ package action
 
 import (
 	"os"
+	"os/exec"
 	"path/filepath"
+	"strings"
 
 	"github.com/helm/helm/log"
+	"github.com/helm/helm/manifest"
 )
 
+// kubeGet executes the kubectl command, override in tests
+var kubeGet func(string) string
```

Comment:
I could see doing things that way. I guess my main objection to that would be that we don't really want layers _above_ Remove to be required to know the internals of how Remove works. So I don't think passing the function into Remove would be a good solution.

My read of what @adamreese is doing here is just making it possible to override the external call for the purposes of unit testing. This does feel a little elaborate if that is the only actual use case... but I don't have a better solution.

I'd still reference the way the HTTP library handles this as generally acceptable for setting a default handler for something.